packages/app/src/specs/RunStatusDots.vue

```diff
@@ -131,8 +142,20 @@ const props = defineProps<{
   specFileExtension: string
 }>()
 
+const emits = defineEmits<{
+  (eventName: 'hasRuns', value: boolean): void
+}>()
+
 const runs = computed(() => {
-  return props.gql?.data?.__typename === 'CloudProjectSpec' ? props.gql?.data?.specRuns?.nodes ?? [] : []
+  const runs = props.gql?.data?.__typename === 'CloudProjectSpec' ? props.gql?.data?.specRuns?.nodes ?? [] : []
+
+  emits('hasRuns', !!runs)
```

Comment:
Good catch. Now that you called it to my attention, I did see it mentioned in the Vue docs to avoid side effects in `computed` properties.  I think I did get a linting hint in another place when trying to change an external value inside a computed property.  I will fix.